### PR TITLE
Donation

### PR DESCRIPTION
**Pull request by babygoat**

### CODE REVIEW

File: controllers/donation.go [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+			for _, errField := range e {
+				failData[errField.Name] = "cannot be empty"
+			}
+		default:
+			// Omit intentionally
+		}
+		return http.StatusBadRequest, gin.H{"status": "fail", "data": failData}, nil
+	}
+
+	userID, _ := strconv.ParseUint(c.Param("userID"), 10, strconv.IntSize)
```
Comment (nickhsine) [inline controllers/donation.go]: no error handling ?

Comment (babygoat) [reply controllers/donation.go]: I'd like to include middlewares.ValidateUserID() later for validation.

**Comment by nickhsine** [comment]
LGTM 👍

File: controllers/donation.go [hunk 2]
```diff
@@ -368,15 +368,21 @@ func getPayMethodID(payMethod string) int {
...
 	var minResp tapPayMinTransactionResp
 	var resp tapPayTransactionResp
+	var err error
 
-	if err := json.Unmarshal(body, &minResp); nil != err {
+	if err = json.Unmarshal(body, &minResp); nil != err {
 		return tapPayTransactionResp{}, errors.New("Cannot unmarshal json response from tap pay server")
 	}
 
+	if 0 != minResp.Status {
```
Comment (nickhsine) [inline controllers/donation.go]: `const tappayOKStatus = 0`?

Comment (babygoat) [reply controllers/donation.go]: done

**Pull request merged**